app/components/post_add_channel_member/post_add_channel_member.js

```diff
@@ -150,43 +152,75 @@ export default class PostAddChannelMember extends React.PureComponent {
+        if (noGroupsUsernames.length) {
+            outOfGroupsMessageID = t('post_body.check_for_out_of_channel_groups_mentions.message');
+            outOfGroupsMessageText = 'did not get notified by this mention because they are not in the channel. They are also not a member of the groups linked to this channel.';
+        }
 
-        return (
-            <Text>
-                {atMentions}
-                {' '}
-                <FormattedText
-                    id={messageId}
-                    defaultMessage={messageText}
-                    style={baseTextStyle}
-                />
-                <Text
-                    style={this.props.textStyles.link}
-                    id='add_channel_member_link'
-                    onPress={this.handleAddChannelMember}
-                >
+        let outOfChannelMessage = null;
+        if (usernames.length) {
+            outOfChannelMessage = (
+                <Text>
+                    {outOfChannelAtMentions}
+                    {' '}
                     <FormattedText
-                        id={linkId}
-                        defaultMessage={linkText}
+                        id={outOfChannelMessageID}
+                        defaultMessage={outOfChannelMessageText}
+                        style={baseTextStyle}
+                    />
+                    <Text
+                        style={this.props.textStyles.link}
+                        id='add_channel_member_link'
+                        onPress={this.handleAddChannelMember}
+                    >
+                        <FormattedText
+                            id={linkId}
+                            defaultMessage={linkText}
+                        />
+                    </Text>
+                    <FormattedText
+                        id={'post_body.check_for_out_of_channel_mentions.message_last'}
+                        defaultMessage={'? They will have access to all message history.'}
+                        style={baseTextStyle}
                     />
                 </Text>
-                <FormattedText
-                    id={'post_body.check_for_out_of_channel_mentions.message_last'}
-                    defaultMessage={'? They will have access to all message history.'}
-                    style={baseTextStyle}
-                />
-            </Text>
+            );
+        }
+
+        let outOfGroupsMessage = null;
+        if (noGroupsUsernames.length) {
+            outOfGroupsMessage = (
+                <Text>
+                    {outOfGroupsAtMentions}
+                    {' '}
+                    <FormattedText
+                        id={outOfGroupsMessageID}
+                        defaultMessage={outOfGroupsMessageText}
+                        style={baseTextStyle}
+                    />
+                </Text>
+            );
+        }
+
+        return (
+            <>
```

Comment:
what is this empty `<>`  tag? I think is safer to use `<React.Fragment>` instead.

Reply:
@enahum `<>` is the short form of `<React.Fragment>`.

Reply:
you always learn something new ;) thanks my friend.

In any case please post a screenshot to see if possible without the groups stuff enabled.

Reply:
Today I Learned... :D